### PR TITLE
Fix #1464, and issue a warning for a SecureRandom new created and used only once

### LINKED ISSUE #1464: Should issue warning for SecureRandom object  created and used only once
I found a false negative when spotbugs-maven-plugin checked [the following line in eclipse/jetty.project](https://github.com/eclipse/jetty.project/blob/jetty-10.0.x/jetty-keystore/src/main/java/org/eclipse/jetty/keystore/KeystoreGenerator.java#L59).

```java
BigInteger serial = BigInteger.valueOf(new SecureRandom().nextLong());
```

According to the description of [DMI: Random object created and used only once (DMI_RANDOM_USED_ONLY_ONCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dmi-random-object-created-and-used-only-once-dmi-random-used-only-once):

> You should strongly consider using a `java.security.SecureRandom` instead (and **avoid allocating a new SecureRandom for each random number needed**).

Spotbugs should generate a warning because the code line `BigInteger serial = BigInteger.valueOf(new SecureRandom().nextLong());` created and used SecureRandom object only once. As noted in the bug description, it should **"avoid allocating a new SecureRandom for each random number"**. However, in the demo, this is allowed by Spotbugs without giving any warning so I think that this is a false negative.

The plugin version used is as follow:

```xml
            <plugin>
                <groupId>com.github.spotbugs</groupId>
                <artifactId>spotbugs-maven-plugin</artifactId>
                <version>4.1.4</version>
                <configuration>
                    <effort>Default</effort>
                    <threshold>Low</threshold>
                </configuration>
                <dependencies>
                    <dependency>
                        <groupId>com.github.spotbugs</groupId>
                        <artifactId>spotbugs</artifactId>
                        <version>4.1.4</version>
                    </dependency>
                </dependencies>
            </plugin>
```

### Demo

Here is a small demo that could be used as test case. See https://github.com/spotbugs/spotbugs/pull/1475

```java
import java.security.SecureRandom;
import java.util.Random;
import edu.umd.cs.findbugs.annotations.ExpectWarning;

public class DMI_RANDOM_USED_ONLY_ONCE {
    @ExpectWarning("DMI_RANDOM_USED_ONLY_ONCE")
    long m1(){
        return new SecureRandom().nextLong();  // This test case will fail now
    }

    @ExpectWarning("DMI_RANDOM_USED_ONLY_ONCE")
    long m2(){
        return new Random().nextLong();  // This will pass
    }
}
```

### PR DESCRIPTION
**Pull request by gloNelson**
Hi,
Fixes #1464 
Merge and upload test-cases present in #1475.
Please help me review @KengoTODA, thanks :)

----
Make sure these boxes are checked before submitting your PR -- thank you!

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

**Review by KengoTODA — APPROVED** [review]
LGTM, I'll wait for other members' review to merge this.

File: spotbugs/src/main/java/edu/umd/cs/findbugs/detect/DumbMethods.java [hunk 1]
```diff
@@ -490,14 +490,16 @@ public void initMethod(Method method) {
 
         @Override
         public void sawOpcode(int seen) {
-            if (seen == Const.INVOKEVIRTUAL && "java/util/Random".equals(getClassConstantOperand())
+            if (seen == Const.INVOKEVIRTUAL && ("java/util/Random".equals(getClassConstantOperand()) || "java/security/SecureRandom".equals(
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/detect/DumbMethods.java:493]: Perhaps `getClassConstantOperand()` should be extracted to a local variable.

**Review by ThrawnCA — APPROVED** [review]

**Comment by ThrawnCA** [comment]
I've approved, but with a suggestion for possible improvement.

File: spotbugs/src/main/java/edu/umd/cs/findbugs/detect/DumbMethods.java (hunk 1 above)
Comment (gloNelson) [reply spotbugs/src/main/java/edu/umd/cs/findbugs/detect/DumbMethods.java]: Done. Thanks for your good suggestion :)

**Review by ThrawnCA — APPROVED** [review]

**Pull request merged**